review/js/script.js

```diff
@@ -1,167 +1,167 @@
+
+let carousel = document.querySelector(".carousel");
+let slider = document.querySelector(".carousel__slider");
+let prevButton = document.querySelector(".carousel__button--prev");
+let nextButton = document.querySelector(".carousel__button--next");
+let indexs = document.querySelectorAll(".carousel__index");
+
+let isOpen = false;
+
+let index = 0;
+let target = 0;
+let current = 0;
+let ease = 0.04;
+let timing = 0;
+
+// interpolazione lineare: utilizzata per calcolare i vari punti a partire da (a,b)
+const lerp = function (start, end, timing) {
+    return start * (1 - timing) + end * timing;
+};
+
+// funzione che si occupa dell'animazione (smooth) del carousel
+const animate = function (dir, width, index) {
+    current = lerp(current, target, ease);
+    current = parseFloat(parseFloat(current).toFixed(2));
+
+    target = width * index;
+
+    slider.style.transform = `translateX(-${Math.round(current)}px)`;
+    timing = requestAnimationFrame(() => animate(dir, width, index));
+};
+
+// funzione che si occupa della gestione delle classi nei bottoni del carousel
+const manageButtons = function () {
+    if (index === 0) {
+        prevButton.classList.add("carousel__button--disabled");
+        nextButton.classList.remove("carousel__button--disabled");
+    } else if (index === slider.children.length - 1) {
+        nextButton.classList.add("carousel__button--disabled");
+        prevButton.classList.remove("carousel__button--disabled");
+    } else {
+        nextButton.classList.remove("carousel__button--disabled");
+        prevButton.classList.remove("carousel__button--disabled");
+    }
+};
+
+// funzione che si occupa della gestione del click sui bottoni
+const handleClick = function (event, dir) {
+    if (isOpen) {
+        const width = slider.children[0].getBoundingClientRect().width;
+        indexs.forEach((node) => node.classList.remove("carousel__index--active"));
+        cancelAnimationFrame(timing);
+
+        if (dir === -1) {
+            index === 0 ? (index = 0) : index--;
+            manageButtons();
+            indexs[index].classList.add("carousel__index--active");
+            return (() => animate(dir, width, index))();
+        } else {
+            index === slider.children.length - 1 ? (index = slider.children.length - 1) : index++;
+            manageButtons();
+            indexs[index].classList.add("carousel__index--active");
+            return (() => animate(dir, width, index))();
+        }
+    }
+};
+
+manageButtons();
+prevButton.addEventListener("click", (event) => handleClick(event, (dir = -1)));
+nextButton.addEventListener("click", (event) => handleClick(event, (dir = 1)));
+
+// gestione dei piccoli dots accanto ai bottoni (gli ho chiamati indici perchè di base sono quelli 😁)
+indexs.forEach((node) =>
+    node.addEventListener("click", (event) => {
+        cancelAnimationFrame(timing);
+        indexs.forEach((node) => node.classList.remove("carousel__index--active"));
```

Comment:
per evitare questo ciclo potremmo tenere traccia in una variabile l'elemento attivo e poi rimuovere la classe una volta che un altra slide viene selezionata